app/serializers/api/v3/ecf/participant_serializer.rb

```diff
@@ -0,0 +1,114 @@
+            if latest_induction_record&.training_status_withdrawn?
+              latest_participant_profile_state = profile.participant_profile_states.sort { |a, b| b.created_at <=> a.created_at }.find { |pps| pps.state == ParticipantProfileState.states[:withdrawn] && pps.cpd_lead_provider_id == cpd_lead_provider.id }
+              if latest_participant_profile_state.present?
+                {
+                  reason: latest_participant_profile_state.reason,
+                  date: latest_participant_profile_state.created_at.rfc3339,
+                }
+              end
+            end
+          end
+
+          def deferral(profile:, cpd_lead_provider:, latest_induction_record:)
+            if latest_induction_record&.training_status_deferred?
+              latest_participant_profile_state = profile.participant_profile_states.sort { |a, b| b.created_at <=> a.created_at }.find { |pps| pps.state == ParticipantProfileState.states[:deferred] && pps.cpd_lead_provider_id == cpd_lead_provider.id }
+              if latest_participant_profile_state.present?
+                {
+                  reason: latest_participant_profile_state.reason,
+                  date: latest_participant_profile_state.created_at.rfc3339,
+                }
+              end
+            end
+          end
+
+          def participant_status(induction_record:)
+            if induction_record.end_date.present?
+              induction_record.end_date > Time.zone.now ? "leaving" : "left"
+            elsif induction_record.start_date > Time.zone.now
+              "joining"
+            else
+              case induction_record.induction_status
+              when "active", "completed"
+                "active"
+              when "withdrawn", "changed"
+                "withdrawn"
+              end
+            end
+          end
+        end
+
+        set_id :id
+
+        set_type :participant
+
+        attribute :full_name
+
+        attribute(:teacher_reference_number) do |object|
+          object.teacher_profile&.trn
+        end
+
+        attribute(:updated_at) do |object|
+          object.updated_at.rfc3339
+        end
+
+        attribute(:ecf_enrolments) do |object, params|
+          object.participant_profiles.select { |pp| [ParticipantProfile::ECT.name, ParticipantProfile::Mentor.name].include?(pp.type) }.map { |profile|
+            latest_induction_record = profile.induction_records.first
+
+            next unless params[:cpd_lead_provider] && latest_induction_record.present? && latest_induction_record.induction_programme&.partnership&.lead_provider&.cpd_lead_provider == params[:cpd_lead_provider]
+
+            {
+              training_record_id: profile.id,
+              email: latest_induction_record.preferred_identity&.email.presence || object.email,
+              mentor_id: latest_induction_record.participant_profile.ect? ? latest_induction_record.mentor_profile&.participant_identity&.user_id : nil,
```

Comment:
ah I think I spotted some n+1s here will need to address as well (in later pages :))